src/plugins/banner/server/routes/fetch_external_config.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import https from 'https';
+import http from 'http';
+import { URL } from 'url';
+import { BannerConfig } from '../../common';
+
+/**
+ * Fetches banner configuration from an external URL
+ * @param url The URL to fetch the banner configuration from
+ * @param logger Logger instance for logging errors
+ * @returns The banner configuration or null if there was an error
+ */
+export async function fetchExternalConfig(
+  url: string,
+  logger: { error: (message: string) => void }
+): Promise<Partial<BannerConfig> | null> {
+  return new Promise((resolve) => {
+    try {
+      const parsedUrl = new URL(url);
+      const requestModule = parsedUrl.protocol === 'https:' ? https : http;
+
+      const req = requestModule.request(
```

Comment:
Can we use fetch() in Node 18+ (or node-fetch), which handles timeouts, errors, and streaming internally?